docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,225 @@
+---
+title: Neat-enhancement-idea
+authors:
+  - "@ishitasequeira" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all intereste parties here.
+reviewers:
+  - "@jannfis"
+  - TBD
+approvers:
+  - "@jannfis"
+  - TBD
+
+creation-date: 2022-01-28
+last-updated: 2022-01-28
+---
+
+# Multiple Sources for application
+
+Support more than one source for creating an Application.
+
+Related Issues: 
+* https://github.com/argoproj/argo-cd/issues/677
+* https://github.com/argoproj/argo-cd/issues/2789
+
+## Open Questions
+* Adding external sources to the Application reource would add additional latencies for creation/reconcilation process. Should we add it to the doc in Risks?
+* 
+
+## Summary
+
+Currently, Argo CD supports Applications with only a single ApplicationSource. In certain scenarios, it would be useful to support more than one source for the application. For example, consider a need to create multiple applications with the same manifests. Today we would have to copy manifest files from one application to another.
+
+Creating applications from multiple sources would allow users to configure multiple services stored at various sources within the same application.
+
+## Motivation
+
+The main motivation behind this enhancement proposal is to allow users to create an application using services that are stored in various sources.
+
+### Goals
+
+The goals of the enhancement are:
+
+#### **Supporting multiple sources for creating an application**
+
+Users should be able to specify multiple sources to add services to the application. Argo CD should compile all the sources and reconcile each source individually for creating the application.
+
+#### **Allow specifying external value files for Helm repositories**
```

Comment:
IMHO this small section - should be split into 1 feature proposal.. more smaller features - instead of one huge one like this seems - and I see no relation between "external values files in Helm applications" and allowing "multiple sources for creating an application" ?

Reply:
Responded on the issue thread: https://github.com/argoproj/argo-cd/issues/2789#issuecomment-1030235017